dvc/stage/__init__.py

```diff
@@ -60,21 +60,60 @@ def loads_from(cls, repo, path, wdir, data):
 
 
 def create_stage(cls, repo, path, external=False, **kwargs):
+    """Create a stage from the class *cls*
+
+    :param cls: the class from which the stage will be
+                created (e.g Stage, PipelineStage).
+    :param repo: the repository object (Repo class).
+    :param path: the path to the dvc filename. If None,
+                 path become <out>.dvc
+    :param external: if True, prevents users from accidentally
+                     using external outputs.
+                     See https://github.com/iterative/dvc/issues/1545
+                     for more details.
+
+    the method can also receive other params as:
+    :param outs: a list of output file (or dir) names.
+    :param deps: a list of urls that represents the source of files.
+    # NOTE there are other possible parameters.
+
+    :returns: a *cls* object representing the stage.
+    """
     from dvc.dvcfile import check_dvc_filename
 
     wdir = os.path.abspath(kwargs.get("wdir", None) or os.curdir)
-    path = os.path.abspath(path)
-    check_dvc_filename(path)
     check_stage_path(repo, wdir, is_wdir=kwargs.get("wdir"))
+
+    # path is None because we define its filename from the dependency.
+    stage = loads_from(cls, repo, None, wdir, kwargs)
+
+    fill_stage_dependencies(
+        stage, **project(kwargs, ["deps", "erepo", "params"])
+    )
+
+    param_outs = kwargs.get("outs")
+
+    if param_outs:
+        outs = []
+        for i, o in enumerate(param_outs):
+            outs.append(o or stage.deps[i].get_file_name())
```

Comment:
This will require establishing a connection, which will break --no-exec case.

Seems like we are overcomplicating this. If the name from the url is ugly, we could either leave it ugly or user could use `-o` to specify the name he wants. What do you think?

Reply:
It's not a matter of being ugly. Url in the format `gdrive://ID` doesn't provide the name of the file/directory at all. As the unique way to get it is establishing a connection, I don't consider we are overcomplicating. However, if the connection should be avoided, the code can either make `-o` mandatory for this case or use the ID as the file/directory name.

Reply:
Sure, I understand that.

> However, if the connection should be avoided, the code can either make -o mandatory for this case or use the ID as the file/directory name.

Let's not make the behavior dependant on whether or not we have a connection, that would be against the regular dvc behavior and will be a hell of a thing to maintain and explain to people. Let's just keep it simple: unless -o is provided, parse the name from the URL, no matter if it is gdrive://ID or gdrive://ID/some/path.

Reply:
No problem. I'll change it and revert other changes, including those in `stage.create_stage` and `utils.resolve_paths`.

Regarding the build / tests, could you check the message above I'm reporting the errors I'm getting?

Reply:
> Regarding the build / tests, could you check the message above I'm reporting the errors I'm getting?

@julsal The message you've posted doesn't say anything about the tests that are failing, it only says that they are indeed failing. There should be a verbose log above it for each test that has failed.